sn_networking/src/record_store.rs

```diff
@@ -1461,199 +1412,205 @@ mod tests {
-        // Using `times_of_earned` to reflect chunks hit the group.
-        // This can avoid `replication counts` causing mis-understanding.
-        // (number_of_nodes, times_of_earned)
-        let mut distribution_map: BTreeMap<u8, (usize, usize)> = Default::default();
-
-        for (peer_id, stats) in peers.iter() {
-            let leading_byte = NetworkAddress::from_peer(*peer_id)
-                .as_kbucket_key()
-                .hashed_bytes()[0];
-            let entry = distribution_map.entry(leading_byte).or_insert((0, 0));
-            entry.0 += 1;
-            entry.1 += stats.2;
-        }
-
-        for (leading_byte, stats) in distribution_map.iter() {
-            println!("{leading_byte:08b}\t{}\t{} ", stats.0, stats.1)
-        }
     }
 
-    // After the completion of 0 with 2000 chunks, there is still 875 nodes earned nothing
-    // After the completion of 1 with 2000 chunks, there is still 475 nodes earned nothing
-    // After the completion of 2 with 2000 chunks, there is still 314 nodes earned nothing
-    // After the completion of 3 with 2000 chunks, there is still 218 nodes earned nothing
-    // ... ...
-    // After the completion of 115 with 2000 chunks, there is still 56 nodes earned nothing
-    // After the completion of 116 with 2000 chunks, there is still 56 nodes earned nothing
-    // After the completion of 117 with 2000 chunks, there is still 56 nodes earned nothing
-    // After the completion of 118 with 2000 chunks, there is still 56 nodes earned nothing
-    // After the completion of 119 with 2000 chunks, there is still 56 nodes earned nothing
-    // After the completion of 120 with 2000 chunks, there is still 56 nodes earned nothing
-    // After the completion of 121 with 2000 chunks, there is still 56 nodes earned nothing
-    fn pick_cheapest_payee(
-        peers_in_close: &Vec<PeerId>,
-        peers: &HashMap<PeerId, (usize, u64, usize)>,
-    ) -> PeerId {
-        let mut payee = None;
-        let mut cheapest_cost = u64::MAX;
-
-        for peer in peers_in_close {
-            if let Some(stats) = peers.get(peer) {
-                let store_cost = calculate_cost_for_records(&QuotingMetrics {
-                    close_records_stored: stats.0,
-                    max_records: MAX_RECORDS_COUNT,
-                    received_payment_count: stats.2,
-                    live_time: 0,
-                });
-                if store_cost < cheapest_cost {
-                    cheapest_cost = store_cost;
-                    payee = Some(*peer);
-                }
-            } else {
-                panic!("Cannot find stats of {peer:?}");
-            }
-        }
-
-        if let Some(peer_id) = payee {
-            peer_id
-        } else {
-            panic!("Cannot find cheapest payee among {peers_in_close:?}");
-        }
+    fn pick_cheapest_payee(peers: &[PeerStats], close_group: &[usize]) -> usize {
```

Comment:
ideally, shall use the same `network::get_fees_from_store_cost_responses` function

Reply:
hmm, good shout. This algo is wayyy less fair than just choosing lowest cost + no tiebreak atm... 

I guess as we pass in sorted peers (from the test), so the lowest index is effectively random on the chunk store addr. 

But then when we compare addresses, we actually end up always choosing the same... 

Hmmm. If the algo is just "random", that may actually be fine (i mean here, we dont rely on the order passed in, but not doing anything deterministic because we dont need to?)